src/main/java/com/aws/greengrass/deployment/converter/DeploymentDocumentConverter.java

```diff
@@ -201,6 +205,36 @@ private static String parseGroupNameFromConfigurationArn(Configuration config) {
         return groupName;
     }
 
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    private static void convertThingGroupArns(DeploymentDocument.DeploymentDocumentBuilder deployDocBuilder,
+                                              Configuration config) throws InvalidRequestException {
+        if (Utils.isNotEmpty(config.getOnBehalfOf())) {
+            try {
+                // IoT thingGroupArn only gives thing group name without 'thinggroup/' prefix
+                String groupName = Arn.fromString(config.getOnBehalfOf()).resource().resource();
+                if (groupName != null) {
+                    groupName = new StringBuilder(THING_GROUP_RESOURCE_NAME_PREFIX).append(groupName).toString();
+                }
+                deployDocBuilder.onBehalfOf(groupName);
+            } catch (IllegalArgumentException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 232.